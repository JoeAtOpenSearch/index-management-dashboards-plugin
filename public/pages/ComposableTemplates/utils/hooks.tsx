/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */
import { CoreStart } from "opensearch-dashboards/public";
import { CommonService } from "../../../services";
import { TemplateItem, TemplateItemRemote } from "../../../../models/interfaces";

export const getTemplate = async (props: {
  templateName: string;
  commonService: CommonService;
  coreService: CoreStart;
}): Promise<TemplateItem> => {
  const response = await props.commonService.apiCaller<{
    index_templates: { name: string; index_template: TemplateItemRemote }[];
  }>({
    endpoint: "transport.request",
    data: {
      method: "GET",
      path: `_index_template/${props.templateName}?flat_settings=true`,
    },
  });
  let error: string = "";
  if (response.ok) {
    const findItem = response.response?.index_templates?.find((item) => item.name === props.templateName);
    if (findItem) {
      const templateDetail = findItem.index_template;
      return JSON.parse(JSON.stringify(templateDetail));
    }
    error = `The template [${props.templateName}] does not exist.`;
  } else {
    error = response.error || "";
  }

  props.coreService.notifications.toasts.addDanger(error);
  throw new Error(error);
};
