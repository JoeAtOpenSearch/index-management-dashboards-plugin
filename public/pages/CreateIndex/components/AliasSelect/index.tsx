/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React, { forwardRef } from "react";
import { EuiComboBoxProps } from "@elastic/eui";
import RemoteSelect, { RemoteSelectProps } from "../../../../components/RemoteSelect";
import { ServerResponse } from "../../../../../server/models/types";

export interface AliasSelectProps extends Omit<EuiComboBoxProps<{ label: string; value: string }>, "value" | "onChange"> {
  value?: Record<string, {}>;
  onChange?: (value: AliasSelectProps["value"]) => void;
  refreshOptions: (aliasName: string) => Promise<ServerResponse<{ alias: string }[]>>;
}

const transformObjToArray = (obj: AliasSelectProps["value"]): { label: string }[] => {
  return Object.keys(obj || {}).map((label) => ({ label }));
};
const transformArrayToObj = (array: { label: string }[]): AliasSelectProps["value"] => {
  return array.reduce((total, current) => ({ ...total, [current.label]: {} }), {});
};

const AliasSelect = forwardRef((props: AliasSelectProps, ref: React.Ref<HTMLInputElement>) => {
  const { value, onChange, refreshOptions: refreshOptionsFromProps } = props;
  const refreshOptions: RemoteSelectProps["refreshOptions"] = ({ searchValue }) => {
    return refreshOptionsFromProps(searchValue || "").then((res) => {
      if (res.ok) {
        return {
          ...res,
          response: [...new Set(res.response.map((item) => item.alias).filter((item) => !item.startsWith(".")))].map((item) => ({
            label: item,
          })),
        };
      } else {
        return res;
      }
    });
  };
  return (
    <RemoteSelect
      isDisabled={props.isDisabled}
      placeholder="Select or create aliases"
      customOptionText="Add {searchValue} as a new alias"
      refreshOptions={refreshOptions}
      value={transformObjToArray(value).map((item) => item.label)}
      onChange={(val) => {
        onChange && onChange(transformArrayToObj(val.map((label) => ({ label }))));
      }}
    />
  );
});

AliasSelect.displayName = "AliasSelect";

// @ts-ignore
export default AliasSelect;
