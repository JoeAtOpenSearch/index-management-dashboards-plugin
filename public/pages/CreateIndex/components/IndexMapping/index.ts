/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import IndexMapping from "./IndexMapping";

export default IndexMapping;
