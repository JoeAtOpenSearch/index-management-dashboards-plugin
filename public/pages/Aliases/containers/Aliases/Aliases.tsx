/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React, { Component, useContext } from "react";
import _ from "lodash";
import { RouteComponentProps } from "react-router-dom";
import queryString from "query-string";
import {
  EuiHorizontalRule,
  EuiBasicTable,
  Criteria,
  EuiTableSortingType,
  Direction,
  Pagination,
  EuiTableSelectionType,
  Query,
} from "@elastic/eui";
import { ContentPanel, ContentPanelActions } from "../../../../components/ContentPanel";
import { DEFAULT_PAGE_SIZE_OPTIONS, DEFAULT_QUERY_PARAMS } from "../../utils/constants";
import CommonService from "../../../../services/CommonService";
import { IAlias } from "../../interface";
import { BREADCRUMBS, ROUTES } from "../../../../utils/constants";
import { getErrorMessage } from "../../../../utils/helpers";
import { CoreServicesContext } from "../../../../components/core_services";
import { ServicesContext } from "../../../../services";
import IndexControls from "../../components/IndexControls";

interface AliasesProps extends RouteComponentProps {
  commonService: CommonService;
}

interface AliasesState {
  totalAliases: number;
  from: string;
  size: string;
  search: string;
  query: Query;
  sortField: keyof IAlias;
  sortDirection: Direction;
  selectedItems: IAlias[];
  aliases: IAlias[];
  loading: boolean;
}

class Aliases extends Component<AliasesProps, AliasesState> {
  static contextType = CoreServicesContext;
  constructor(props: AliasesProps) {
    super(props);
    const {
      from = DEFAULT_QUERY_PARAMS.from,
      size = DEFAULT_QUERY_PARAMS.size,
      search = DEFAULT_QUERY_PARAMS.search,
      sortField = DEFAULT_QUERY_PARAMS.sortField,
      sortDirection = DEFAULT_QUERY_PARAMS.sortDirection,
    } = queryString.parse(location.search) as {
      from: string;
      size: string;
      search: string;
      sortField: keyof IAlias;
      sortDirection: Direction;
    };
    this.state = {
      totalAliases: 0,
      from,
      size,
      search,
      query: Query.parse(search),
      sortField,
      sortDirection,
      selectedItems: [],
      aliases: [],
      loading: false,
    };

    this.getAliases = _.debounce(this.getAliases, 500, { leading: true });
  }

  async componentDidMount() {
    this.context.chrome.setBreadcrumbs([BREADCRUMBS.INDEX_MANAGEMENT, BREADCRUMBS.INDICES]);
    await this.getAliases();
  }

  getQueryState = (state: AliasesState) => {
    return Object.keys(DEFAULT_QUERY_PARAMS).reduce((total, key) => {
      return {
        ...total,
        [key]: state[key as keyof typeof DEFAULT_QUERY_PARAMS],
      };
    }, {} as AliasesState);
  };

  getAliases = async (): Promise<void> => {
    this.setState({ loading: true });
    try {
      const { history, commonService } = this.props;
      const queryObject = this.getQueryState(this.state);
      const queryParamsString = queryString.stringify(queryObject);
      history.replace({ ...this.props.location, search: queryParamsString });

      const getIndicesResponse = await commonService.apiCaller<IAlias[]>({
        endpoint: "cat.aliases",
        data: {
          format: "json",
        },
      });

      if (getIndicesResponse.ok) {
        const aliases: IAlias[] = getIndicesResponse.response.map((item) => ({
          ...item,
          aliasId: `${item.alias}-${item.index}`,
        }));
        const totalAliases = aliases.length;
        const payload = {
          aliases,
          totalAliases,
          selectedItems: this.state.selectedItems
            .map((item) => aliases.find((remoteItem) => remoteItem.index === item.index))
            .filter((item) => item),
        } as AliasesState;
        this.setState(payload);
      } else {
        this.context.notifications.toasts.addDanger(getIndicesResponse.error);
      }
    } catch (err) {
      this.context.notifications.toasts.addDanger(getErrorMessage(err, "There was a problem loading the indices"));
    }

    // Avoiding flicker by showing/hiding the "Data stream" column only after the results are loaded.
    this.setState({ loading: false });
  };

  onTableChange = ({ page: tablePage, sort }: Criteria<IAlias>): void => {
    const { index: page, size } = tablePage || {};
    const { field: sortField, direction: sortDirection } = sort || {};
    this.setState({
      from: "" + page,
      size: "" + size,
      sortField: sortField || DEFAULT_QUERY_PARAMS.sortField,
      sortDirection: sortDirection as Direction,
    });
  };

  onSelectionChange = (selectedItems: IAlias[]): void => {
    this.setState({ selectedItems });
  };

  resetFilters = (): void => {
    this.setState({ search: DEFAULT_QUERY_PARAMS.search, query: Query.parse(DEFAULT_QUERY_PARAMS.search) });
  };

  onSearchChange = ({ query }: { query: string }): void => {
    this.setState({ from: "0", search: query });
    this.getAliases();
  };

  render() {
    const { totalAliases, from, size, sortField, sortDirection, aliases } = this.state;

    const pagination: Pagination = {
      pageIndex: Number(from),
      pageSize: Number(size),
      pageSizeOptions: DEFAULT_PAGE_SIZE_OPTIONS,
      totalItemCount: Number(totalAliases),
    };

    const sorting: EuiTableSortingType<IAlias> = {
      sort: {
        direction: sortDirection,
        field: sortField,
      },
    };

    const selection: EuiTableSelectionType<IAlias> = {
      onSelectionChange: this.onSelectionChange,
    };
    return (
      <ContentPanel
        actions={
          <ContentPanelActions
            actions={[
              {
                text: "Create Alias",
                buttonProps: {
                  fill: true,
                  onClick: () => {
                    this.props.history.push(ROUTES.CREATE_INDEX);
                  },
                },
              },
            ]}
          />
        }
        bodyStyles={{ padding: "initial" }}
        title="Aliases"
      >
        <IndexControls search={this.state.search} onSearchChange={this.onSearchChange} />
        <EuiHorizontalRule margin="xs" />

        <EuiBasicTable
          columns={[
            {
              field: "alias",
              name: "Alias Name",
              sortable: true,
            },
            {
              field: "index",
              name: "Index Name",
              sortable: true,
            },
          ]}
          isSelectable={true}
          itemId="aliasId"
          items={aliases}
          onChange={this.onTableChange}
          pagination={pagination}
          selection={selection}
          sorting={sorting}
        />
      </ContentPanel>
    );
  }
}

export default function AliasContainer(props: Omit<AliasesProps, "commonService">) {
  const context = useContext(ServicesContext);
  return <Aliases {...props} commonService={context?.commonService as CommonService} />;
}
