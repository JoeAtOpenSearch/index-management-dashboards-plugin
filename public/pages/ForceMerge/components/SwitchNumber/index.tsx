import React, { useState } from "react";
import { EuiRadioGroup, EuiSpacer } from "@elastic/eui";
import CustomFormRow from "../../../../components/CustomFormRow";
import { AllBuiltInComponents } from "../../../../components/FormGenerator";

interface SwitchNumberProps {
  value?: number;
  onChange: (val: SwitchNumberProps["value"]) => void;
}

export default function SwitchNumber(props: SwitchNumberProps) {
  const [id, setId] = useState(props.value && props.value > 0 ? "1" : "0");
  return (
    <>
      <EuiRadioGroup
        options={[
          {
            id: "0",
            label: "Automatically",
          },
          {
            id: "1",
            label: "Manually set",
          },
        ]}
        idSelected={id}
        onChange={(id) => {
          setId(id);
          if (id === "0") {
            props.onChange(undefined);
          } else {
            props.onChange(1);
          }
        }}
      />
      {id === "1" ? (
        <>
          <EuiSpacer size="s" />
          <CustomFormRow helpText="Specify the number of segments to merge to">
            <AllBuiltInComponents.Number {...props} />
          </CustomFormRow>
        </>
      ) : null}
    </>
  );
}
