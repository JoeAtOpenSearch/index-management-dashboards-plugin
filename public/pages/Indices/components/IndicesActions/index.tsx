/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */
import React, { useCallback, useContext, useMemo, useState } from "react";
import {
  // @ts-ignore
  Criteria,
  // @ts-ignore
  Pagination,
  EuiButton,
  EuiContextMenu,
} from "@elastic/eui";

import { ManagedCatIndex } from "../../../../../server/models/interfaces";
import { ServerResponse } from "../../../../../server/models/types";
import ApplyPolicyModal from "../../components/ApplyPolicyModal";
import SimplePopover from "../../../../components/SimplePopover";
import { ModalConsumer } from "../../../../components/Modal";
import { CoreServicesContext } from "../../../../components/core_services";
import DeleteIndexModal from "../../components/DeleteIndexModal";
import { ServicesContext } from "../../../../services";

export interface IndicesActionsProps {
  selectedItems: ManagedCatIndex[];
  onDelete: (indices: string) => Promise<ServerResponse<any>>;
}

export default function IndicesActions(props: IndicesActionsProps) {
  const { selectedItems, onDelete } = props;
  const [deleteIndexModalVisible, setDeleteIndexModalVisible] = useState(false);
  const context = useContext(CoreServicesContext);
  const services = useContext(ServicesContext);

  const onDeleteIndexModalClose = () => {
    setDeleteIndexModalVisible(false);
  };

  const onDeleteIndexModalConfirm = useCallback(async () => {
    const result = await onDelete(selectedItems.map((item) => item.index).join(","));
    if (result && result.ok) {
      onDeleteIndexModalClose();
    }
  }, [selectedItems, onDelete, services, context, onDeleteIndexModalClose]);

  const renderKey = useMemo(() => Date.now(), [selectedItems]);

  return (
    <>
      <ModalConsumer>
        {({ onShow }) => (
          <SimplePopover
            data-test-subj="More Action"
            panelPaddingSize="none"
            button={
              <EuiButton iconType="arrowDown" iconSide="right">
                Actions
              </EuiButton>
            }
          >
            <EuiContextMenu
              initialPanelId={0}
              // The EuiContextMenu has bug when testing in jest
              // the props change won't make it rerender
              key={renderKey}
              panels={[
                {
                  id: 0,
                  items: [
                    {
                      name: "Delete",
                      disabled: !selectedItems.length,
                      "data-test-subj": "Delete Action",
                      onClick: () => setDeleteIndexModalVisible(true),
                    },
                    {
                      name: "Apply policy",
                      disabled: !selectedItems.length,
                      "data-test-subj": "Apply policyButton",
                      onClick: () =>
                        onShow(ApplyPolicyModal, {
                          indices: selectedItems.map((item: ManagedCatIndex) => item.index),
                          core: context,
                        }),
                    },
                  ],
                },
              ]}
            />
          </SimplePopover>
        )}
      </ModalConsumer>
      <DeleteIndexModal
        selectedItems={selectedItems.map((item) => item.index)}
        visible={deleteIndexModalVisible}
        onClose={onDeleteIndexModalClose}
        onConfirm={onDeleteIndexModalConfirm}
      />
    </>
  );
}
