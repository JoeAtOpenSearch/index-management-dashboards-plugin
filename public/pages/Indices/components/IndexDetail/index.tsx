/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React, { useState } from "react";
import { EuiButtonEmpty, EuiCopy } from "@elastic/eui";
import { IndexItem } from "../../../../../models/interfaces";
import { Modal } from "../../../../components/Modal";
import IndicesActions from "../IndicesActions";
import { ManagedCatIndex } from "../../../../../server/models/interfaces";

export interface IndexDetailModalProps {
  index: string;
  onDelete?: () => void;
  record: ManagedCatIndex;
}

export default function IndexDetail(props: IndexDetailModalProps) {
  // const { detail, visible } = props;
  const { index, record, onDelete } = props;
  const [visible, setVisible] = useState(false);
  const [detail, setDetail] = useState({} as IndexItem);
  return (
    <>
      <EuiCopy textToCopy={index}>
        {(copy) => <EuiButtonEmpty size="xs" flush="right" iconType="copyClipboard" onClick={copy} color="text"></EuiButtonEmpty>}
      </EuiCopy>
      <EuiButtonEmpty onClick={() => setVisible(true)} data-test-subj={`view-index-detail-button-${index}`}>
        {index}
      </EuiButtonEmpty>
      <Modal.SimpleModal
        visible={visible}
        maxWidth={false}
        style={{
          width: "60vw",
        }}
        onClose={() => setVisible(false)}
        title={
          <div style={{ display: "flex", justifyContent: "space-between" }}>
            <span>{index}</span>
            <IndicesActions selectedItems={[record]} onDelete={onDelete} />
          </div>
        }
      />
    </>
  );
}
