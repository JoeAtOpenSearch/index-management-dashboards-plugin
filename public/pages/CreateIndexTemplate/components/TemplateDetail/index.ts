/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import IndexDetail from "./TemplateDetail";

export default IndexDetail;
export * from "./TemplateDetail";
