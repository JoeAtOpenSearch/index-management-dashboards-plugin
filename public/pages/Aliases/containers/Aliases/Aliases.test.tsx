/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { render } from "@testing-library/react";
import { Redirect, Route, Switch } from "react-router-dom";
import { HashRouter as Router } from "react-router-dom";
import { browserServicesMock, coreServicesMock } from "../../../../../test/mocks";
import Aliases from "./Aliases";
import { ServicesContext } from "../../../../services";
import { ROUTES } from "../../../../utils/constants";
import { CoreServicesContext } from "../../../../components/core_services";

function renderWithRouter() {
  return {
    ...render(
      <Router>
        <Switch>
          <Route
            path={ROUTES.ALIASES}
            render={(props) => (
              <CoreServicesContext.Provider value={coreServicesMock}>
                <ServicesContext.Provider value={browserServicesMock}>
                  <Aliases {...props} />
                </ServicesContext.Provider>
              </CoreServicesContext.Provider>
            )}
          />
          <Redirect from="/" to={ROUTES.ALIASES} />
        </Switch>
      </Router>
    ),
  };
}

describe("<Indices /> spec", () => {
  it("renders the component", async () => {
    const { container } = renderWithRouter();

    expect(container.firstChild).toMatchSnapshot();
  });
});
