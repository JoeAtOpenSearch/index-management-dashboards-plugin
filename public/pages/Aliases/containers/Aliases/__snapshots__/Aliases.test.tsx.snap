// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Indices /> spec renders the component 1`] = `
<div
  class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow"
  style="padding-left: 0px; padding-right: 0px;"
>
  <div
    class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsFlexStart euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow euiFlexGroup--responsive"
    style="padding: 0px 10px;"
  >
    <div
      class="euiFlexItem"
    >
      <h3
        class="euiTitle euiTitle--large"
      >
        Aliases
      </h3>
    </div>
    <div
      class="euiFlexItem euiFlexItem--flexGrowZero"
    >
      <div
        class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow euiFlexGroup--responsive"
      >
        <div
          class="euiFlexItem"
        >
          <div
            class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow euiFlexGroup--responsive"
          >
            <div
              class="euiFlexItem euiFlexItem--flexGrowZero"
            >
              <button
                class="euiButton euiButton--primary euiButton--fill"
                data-test-subj="Create AliasButton"
                type="button"
              >
                <span
                  class="euiButtonContent euiButton__content"
                >
                  <span
                    class="euiButton__text"
                  >
                    Create Alias
                  </span>
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <hr
    class="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginXSmall"
  />
  <div
    style="padding: initial;"
  >
    <div
      class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow euiFlexGroup--responsive"
      style="padding: 0px 5px;"
    >
      <div
        class="euiFlexItem"
      >
        <div
          class="euiFlexGroup euiFlexGroup--gutterMedium euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow euiFlexGroup--responsive euiFlexGroup--wrap"
        >
          <div
            class="euiFlexItem euiSearchBar__searchHolder"
          >
            <div
              class="euiFormControlLayout euiFormControlLayout--fullWidth"
            >
              <div
                class="euiFormControlLayout__childrenWrapper"
              >
                <input
                  aria-label="This is a search bar. After typing your query, hit enter to filter the results lower in the page."
                  class="euiFieldSearch euiFieldSearch--fullWidth"
                  placeholder="Search..."
                  type="search"
                  value=""
                />
                <div
                  class="euiFormControlLayoutIcons"
                >
                  <span
                    class="euiFormControlLayoutCustomIcon"
                  >
                    EuiIconMock
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <hr
      class="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginXSmall"
    />
    <div
      class="euiBasicTable"
    >
      <div>
        <div
          class="euiTableHeaderMobile"
        >
          <div
            class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsBaseline euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow"
          >
            <div
              class="euiFlexItem euiFlexItem--flexGrowZero"
            >
              <div
                class="euiCheckbox"
              >
                <input
                  aria-label="Select all rows"
                  class="euiCheckbox__input"
                  disabled=""
                  id="_selection_column-checkbox_some_html_id"
                  type="checkbox"
                />
                <div
                  class="euiCheckbox__square"
                />
                <label
                  class="euiCheckbox__label"
                  for="_selection_column-checkbox_some_html_id"
                >
                  Select all rows
                </label>
              </div>
            </div>
            <div
              class="euiFlexItem euiFlexItem--flexGrowZero"
            >
              <div
                class="euiTableSortMobile"
              >
                <div
                  class="euiPopover euiPopover--anchorDownRight"
                >
                  <div
                    class="euiPopover__anchor"
                  >
                    <button
                      class="euiButtonEmpty euiButtonEmpty--primary euiButtonEmpty--xSmall euiButtonEmpty--flushRight"
                      type="button"
                    >
                      <span
                        class="euiButtonContent euiButtonContent--iconRight euiButtonEmpty__content"
                      >
                        EuiIconMock
                        <span
                          class="euiButtonEmpty__text"
                        >
                          Sorting
                        </span>
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <table
          class="euiTable euiTable--responsive"
          id="some_html_id"
          tabindex="-1"
        >
          <caption
            class="euiScreenReaderOnly euiTableCaption"
          />
          <thead>
            <tr>
              <th
                class="euiTableHeaderCellCheckbox"
                scope="col"
              >
                <div
                  class="euiTableCellContent"
                >
                  <div
                    class="euiCheckbox euiCheckbox--inList euiCheckbox--noLabel"
                  >
                    <input
                      aria-label="Select all rows"
                      class="euiCheckbox__input"
                      data-test-subj="checkboxSelectAll"
                      disabled=""
                      id="_selection_column-checkbox_some_html_id"
                      type="checkbox"
                    />
                    <div
                      class="euiCheckbox__square"
                    />
                  </div>
                </div>
              </th>
              <th
                aria-live="polite"
                aria-sort="descending"
                class="euiTableHeaderCell"
                data-test-subj="tableHeaderCell_alias_0"
                role="columnheader"
                scope="col"
              >
                <button
                  class="euiTableHeaderButton euiTableHeaderButton-isSorted"
                  data-test-subj="tableHeaderSortButton"
                  type="button"
                >
                  <span
                    class="euiTableCellContent"
                  >
                    <span
                      class="euiTableCellContent__text"
                      title="Alias Name"
                    >
                      Alias Name
                    </span>
                    EuiIconMock
                  </span>
                </button>
              </th>
              <th
                aria-live="polite"
                aria-sort="none"
                class="euiTableHeaderCell"
                data-test-subj="tableHeaderCell_index_1"
                role="columnheader"
                scope="col"
              >
                <button
                  class="euiTableHeaderButton"
                  data-test-subj="tableHeaderSortButton"
                  type="button"
                >
                  <span
                    class="euiTableCellContent"
                  >
                    <span
                      class="euiTableCellContent__text"
                      title="Index Name"
                    >
                      Index Name
                    </span>
                  </span>
                </button>
              </th>
            </tr>
          </thead>
          <tbody>
            <tr
              class="euiTableRow"
            >
              <td
                class="euiTableRowCell euiTableRowCell--isMobileFullWidth"
                colspan="3"
              >
                <div
                  class="euiTableCellContent euiTableCellContent--alignCenter"
                >
                  <span
                    class="euiTableCellContent__text"
                  >
                    No items found
                  </span>
                </div>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;
